### ISSUE TITLE
Persistence of keys via volumes

### ISSUE THREAD

**Issue by Ramblurr** [eforce21/letsencrypt-nginx-proxy#14]
Currently this image persists the keys and lets encrypt information in the container. This isn't ideal if the container is removed, as key generation for all the hosts must happen again.

Is there a convenient directory in the container that can be mounted as a volume to provide persistence?

If so, that would be a nice addition to the readme.

**Comment by clayrisser** [comment]
From the dockerhub documentation on letsencrypt-nginx-proxy

### Volumes

Letsencrypt saves the certificates in /etc/letsencrypt/live. From there, they are linked to the Nginx certificate folder /etc/nginx/certs. You might want to map either the first volume or both of them, if you like your certifacates to survive container recreation.

**Comment by kortenkamp** [comment]
This is not sufficient, you have to map a volume to `/etc/letsencrypt/archive` as well, as only links to archive will be created in live. Or maybe it is sufficient to map `archive` only – but maybe @eforce21 can comment on that?

**Comment by evodicka** [comment]
You are correct about `/etc/letsencrypt/archive`. This one was missing in the documentation. An additional volume declaration will be added to the Dockerfile of the current development version.
